pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,429 @@
+    * Automatically positions adjacent to the symbol at a fixed position.
+    * Automatically reformats text when the symbol location has changed.
+
+    Parameters
+    ----------
+    target : TargetItem
+        The TargetItem to which this label will be attached to.
+    text : str or callable, Optional
+        Governs the text displayed, can be a fixed string or a format string
+        that accepts the x, and y position of the target item; or be a callable
+        method that accepts a tuple (x, y) and returns a string to be displayed.
+        If None, an empty string is used.  Default is None
+    offset : tuple or list or QPointF or QPoint
+        Position to set the anchor of the TargetLabel away from the center of
+        the target in pixels, by default it is (20, 0).
+    anchor : tuple, list, QPointF or QPoint
+        Position to rotate the TargetLabel about, and position to set the
+        offset value to see :class:`~pyqtgraph.TextItem` for more inforation.
+    kwargs : dict of arguments that are passed on to
+        :class:`~pyqtgraph.TextItem` constructor, excluding text parameter
+    """
+
+    def __init__(
+        self,
+        target,
+        text=None,
+        offset=(20, 0),
+        anchor=(0, 0.5),
+        **kwargs,
+    ):
+        super().__init__(anchor=anchor, **kwargs)
+        self.setParentItem(target)
+        self.target = target
+
+        text = "" if text is None else text
+        self.setFormat(text)
+
+        if isinstance(offset, (tuple, list)):
+            self.offset = Point(*offset)
+        elif isinstance(offset, (QtCore.QPoint, QtCore.QPointF)):
+            self.offset = offset
+        else:
+            raise TypeError("Offset parameter is the wrong data type")
+        self.target.sigPositionChanged.connect(self.valueChanged)
+        self.valueChanged()
+
+    def format(self):
+        return self._format
+
+    def setFormat(self, text):
+        """Method to set how the TargetLabel should display the text.  This
+        method should be called from TargetItem.setLabel directly.
+
+        Parameters
+        ----------
+        text : Callable or str
+            Details how to format the text.
+            If Callable, then the label will display the result of ``text(x, y)``
+            If a fromatted string, then the output of ``text.format(x, y)`` will be
+            displayed
+            If a non-formatted string, then the text label will display ``text``
+        """
+        if not callable(text):
+            parsed = list(string.Formatter().parse(text))
+            if parsed[0][1] is not None:
+                self.setProperty("formattableText", True)
+            else:
+                self.setText(text)
+                self.setProperty("formattableText", False)
+        else:
+            self.setProperty("formattableText", False)
+        self._format = text
+        self.valueChanged()
+
+    def valueChanged(self):
+        x, y = self.target.position()
```

Comment:
This should probably be `.pos()`:
```
  File "examples/InfiniteLine.py", line 45, in <module>
    "offset": QtCore.QPoint(15, 15)
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 120, in __init__
    self.setLabel(label, labelOpts)
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 333, in setLabel
    self._label = TargetLabel(self, text=text, **labelOpts)
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 385, in __init__
    self.setFormat(text)
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 423, in setFormat
    self.valueChanged()
  File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 427, in valueChanged
    x, y = self.target.position()
AttributeError: 'TargetItem' object has no attribute 'position'
```

Reply:
:facepalm: i just renamed that method, of course you would think I would check for any references to said method...